plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestParquetPageSkipping.java

```diff
@@ -114,14 +125,33 @@ public void testPageSkipping(String sortByColumn, String sortByColumnType, Objec
     private int assertSameResults(String query)
     {
         MaterializedResult withColumnIndexing = computeActual(query);
-        MaterializedResult withoutColumnIndexing = computeActual(Session.builder(getSession())
-                        .setCatalogSessionProperty(getSession().getCatalog().get(), "parquet_use_column_index", "false")
-                        .build(),
-                query);
+        MaterializedResult withoutColumnIndexing = computeActual(noColumnIndexFiltering(), query);
         assertEquals(withColumnIndexing, withoutColumnIndexing);
         return withoutColumnIndexing.getRowCount();
     }
 
+    private void assertRowGroupPruning(@Language("SQL") String sql)
+    {
+        DistributedQueryRunner queryRunner = getDistributedQueryRunner();
+        // TODO: replace this with a simple query stats check once we find a way to wait until all pending updates to query stats have been applied
+        // (might be fixed by https://github.com/trinodb/trino/issues/5172)
+        assertEventually(new Duration(10, SECONDS), () -> {
+            ResultWithQueryId<MaterializedResult> withoutColumnIndexing = queryRunner.executeWithQueryId(noColumnIndexFiltering(), sql);
+            QueryStats queryStats = getQueryStats(withoutColumnIndexing.getQueryId());
+            assertThat(queryStats.getPhysicalInputPositions()).isGreaterThan(0);
+            assertThat(queryStats.getProcessedInputPositions()).isEqualTo(queryStats.getPhysicalInputPositions());
+            assertThat(withoutColumnIndexing.getResult().getRowCount()).isEqualTo(0);
+        });
+
+        assertEventually(new Duration(10, SECONDS), () -> {
+            ResultWithQueryId<MaterializedResult> withColumnIndexing = queryRunner.executeWithQueryId(getSession(), sql);
+            QueryStats queryStats = getQueryStats(withColumnIndexing.getQueryId());
```

Comment:
instead of `getQueryStats`, we could have `assertQueryStats` which would  accept`(sesssion, sql, Consumer<QueryStats>)` and do `assertEventually` internally

Reply:
done